src/app/Dashboard/AddCard.tsx

```diff
@@ -630,7 +630,12 @@ const SelectControl = ({ handleChange, control, selectedConfig }: SelectControlP
     >
       {errored
         ? [<SelectOption key={0} value={`Load Error: ${options[0]}`} isPlaceholder isDisabled />]
-        : options.map((choice, idx) => <SelectOption key={idx + 1} value={choice} />)}
+        : options.map((choice, idx) => {
+            if (control.extras && control.extras.displayMapper) {
+              choice = control.extras.displayMapper(choice);
```

Comment:
hey @maxcao13, sorry. I saw the type of the `PropContro.values` as `any[] | Observable<any>`. Do you think we should convert `choice` to string first? Like `${choice}`?